FirebaseSessions/Sources/Settings/SettingsProtocol.swift

```diff
@@ -0,0 +1,29 @@
+//
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// Provides the APIs to access Settings and their configuration values
+protocol SettingsProtocol {
+  // Config to show if sessions is enabled
+  var sessionsEnabled: Bool? { get }
+
+  // Config showing the sampling rate for sessions
+
+  var samplingRate: Double? { get }
+
+  // Background timeout config value before which a new session is generated
+  var sessionTimeout: TimeInterval? { get }
```

Comment:
Now, there may be a reason for this that I'm not aware of, so apologies and disregard if so. But all of this protocol's fields seem like they could be non-optional. For example, if I'm given a type that conforms to this protocol, I would expect that it can be in two states, either enabled or disabled. The third `nil` state is a bit harder to reason (to me, at least) and also adds an extra state that complicates client code. 

I know this protocol was defined in a previous PR but I did want to mention it in case you agree and think it is helpful. I do think reducing the use of optionals to only cases where it is absolutely necessary is a good practice and can eliminate away a lot of unnecessary code paths.

Reply:
This is a good callout. In this case what we're doing is relying on the nil values for when the specified "layer" doesn't have a value to provide. So we're saying we have 3 Settings layers, one fetched, one configured via plist, and the 3rd is the defaults. The defaults always return a value, but the other 2 layers may not have values to provide.

Luckily, this API will be easy to use because the getters on `SessionsSettings` are not optional. The SDK will read values from `SessionsSettings` and not the other layers

Reply:
Very good question and a valid one. I would typically try to be in the same state. The reason for the third state is that there is a prioritization of a config value if the value is available or not. In other words, there is a state which qualifies if a config value is available or not, in addition to if it the config value says enabled/disabled. So, had to go with this approach (a common paradigm for such situations). TBH: I first started my approach without the third state, but that would quite honestly add more complications/less-understandable code.